### DISCUSSION TITLE
Random Flickering + Raspberry Pi Restarts with HyperHDR on SK6812 LED Strip

### DISCUSSION THREAD

**Discussion by gghisellini** [awawa-dev/HyperHDR#624] [Q&A]
Hello, HyperHDR community,

I am reaching out because I've encountered random flickering and unexpected restarts on my Raspberry Pi while using HyperHDR. Any help is appreciated because I don't know what I'm missing here.

Setup Details:
LEDs: RGBW SK6812.

Power Supply: I'm using the Aclorol 5V 10A 50W.

Raspberry Pi 4 Model B (2019 version).

Power & Ground Connections: I've set up the power for the LED strip and the Raspberry Pi using a USB-C cable which I manually split open. I employed a 12V DC Power Connector (5.5mm x 2.1mm, 24V Power Jack Plug Barrel Adapter) to manage the connections. This helps me bridge the power and ground between the LED strip and the Raspberry Pi.

HyperHDR Issue: The HyperHDR interface indicates that an "undervoltage" has been detected., maybe this is the issue?

I'm still not sure how to fix the random flickers on the LED strip and the unexpected restarts of the Raspberry Pi after a few minutes, or even less, of use.
Thank you in advance for your time and assistance!

Here are some pictures of my configs in HyperHDR.
![hyperHDR1](https://github.com/awawa-dev/HyperHDR/assets/102735264/0e7cff1f-1bee-4ed6-b6fb-0b8f9333273a)
![image](https://github.com/awawa-dev/HyperHDR/assets/102735264/fa489c1b-cba7-43a3-ba72-85a99a4ae6b6)
![image](https://github.com/awawa-dev/HyperHDR/assets/102735264/b61916c9-3895-4418-a063-49a1b677c0fd)
![image](https://github.com/awawa-dev/HyperHDR/assets/102735264/f092e2df-0682-450b-9764-e3981014095c)

**Comment by awawa-dev** [comment]
Hi
The driver you are using is not recommended and rather unstable and the risk increases with the number of LEDs and the use of RGBW LEDs. Rpi restarts are disturbing, it is possible that the RPi fuse (or LED power supply protection) is triggered. Assuming that you have connected everything correctly and there is no short circuit, e.g. in a made usb-c adapter, unstable power supply may be the main problem (more [https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#how-to-read-hyperhdr-statistics](https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#how-to-read-hyperhdr-statistics)): RPi reboots and sk6812 flickering (does it happen on solid color set in the remote control tab?). This type of LED by its specification is usually very resistant to voltage fluctuations so they would have to be significant. Rpi needs to be stabilized first before it's too late. For example, you can try to use an independent USB power supply for RPi and provide common ground and/or disconnect LED data cable from GPIO for testing.

**Comment by gghisellini** [comment]
Hello, thank you for your quick response.

What driver would you recommend for these leds?
Would you recommend other pieces of hardware as well?
The Pi is stable and the LEDs do work.
It's just that when I connect the data cable to the GPIO 18 on the Pi, everything starts to crumble and eventually it restarts and all LEDs stay turned on very bright and that's it.
I have to unplug from the current and start all over again.

**Reply by awawa-dev** [reply]
Without LED control, the power supply is not loaded. But did you check to see if there are still low voltage warnings after a while? 10A is dangerously close to the demand limit of your setup. Typically, only very good power supplies can maintain their maximum specifications for a longer time, although they are also susceptible to a drop in maximum performance due to temperature rise.

Before you start modifying the next elements in your setup (which is not yet decided to be necessary), power your Rpi with an independent USB min. 2A power supply, preferably an original one. You must then provide a common ground between the ground of the LED strip and one of RPi GPIO ground. If you are looking for an alternative LED driver check out this thread: https://github.com/awawa-dev/HyperHDR/discussions/561 there are boards on the market with a built-in 3.3V to 5V level shifter that you are also missing (RPi GPIO is only 3.3v).

**Reply by gghisellini** [reply]
Can I contact you outside GitHub so we can talk more fluently?
I totally understand if not possible.

**Reply by awawa-dev** [reply]
In justified cases related to HyperHDR, if they contain things that you do not want to make public, you can contact me by e-mail (it is given in my Github profile). Since English is not my primary language, I prefer to communicate via email or here on the forum, so that other users can benefit from the knowledge in diagnosing and solving similar problems in the future.

**Reply by gghisellini** [reply]
I will post the solution here once we get to it. It's just that this can take a lot of time to debug, and I'd like to have everything compacted in one place to generate a complete solution.

Unfortunately, I could not get your email from your GitHub profile.

Again, thank you so much for your time, and I admire what you do.

**Reply by awawa-dev** [reply]
If we want to do everything in accordance with electrical standards and do not count on luck, then unfortunately the construction will not be that simple. Maybe my last blog post will help you? Since support for Pico boards with a built-in level shifter has been added in the meantime, it can be made even simpler. I would only worry about debugging if a separate power supply for the RPi didn't help.

**Reply by gghisellini** [reply]
I've got a Raspberry Pi Zero WH as well.

Should I try that?

**Reply by awawa-dev** [reply]
No, the Zero is a different design than the simpler and cheaper Pico (rp2040), but the Pico has other unique advantages that come in handy when driving LED strips. As for the main-unit (not the led driver) the RPi 4 you're using is fine and you don't want to downgrade it to Zero.

**Reply by gghisellini** [reply]
The issue with my Pi4 now is that debugging, I plugged directly into the Aclorol 5V 10A 50W, and it started smoking a little bit, and I believe it died.

**Reply by awawa-dev** [reply]
Non-branded Chinese power supplies have a very bad reputation. The polyfuse will usually protect the Rpi well if powered via the USB port (not by GPIO) unless one of the wires of the custom USB-C adapter has been misconnected. If it's blown out and won't reset, it may just need to be replaced. If the RPi pmic has been damaged, it may not be worth repairing.

**Reply by gghisellini** [reply]
If you had to suggest me hardware to get this going as well as possible, what would you recommend to me?
Talking about power supply, main unit, etc?

**Reply by awawa-dev** [reply]
Once I made one post: https://www.hyperhdr.eu/2020/11/hyperhdr-prepare-for-building-buying.html I prefer modular Mean-Well in my setups and have little experience with 5v laptop type power supplies. And the original power supply for the RPi.

**Reply by gghisellini** [reply]
And what main unit would you use? I think my Pi 4 died, I can't power it up.

**Reply by awawa-dev** [reply]
RPi 4, even 1GB version is enough for minimal Raspberry Pi OS when only HyperHDR is running.

**Reply by gghisellini** [reply]
As my Pi4 is dead, I'll try with my Pi Zero. Do you think it will be fine?

**Reply by awawa-dev** [reply]
Barely, depending on expectations: Zero 2 is good, but lacks USB 3.0, Zero 1st generation is far behind. Some results of the old and current generation of Zero: https://www.hyperhdr.eu/2021/10/hands-on-with-new-raspberry-pi-zero-2-w.html

**Reply by awawa-dev** [reply]
Small correction: there is no fuse in the RPi 4 anymore (yes, let's cut costs more, let users have more problems...) only a $0.50 zener diode that protects the RPi 4 from over-voltage. With a lot of luck, it's enough to replace it. Or temporarily remove this diode for the test if we're going to throw it away anyway, which is a pity, many components could survive and are quite valuable..The RPi should boot without this diode if that was the only fault, but eventually it will need to be replaced. Keep in mind that I've never tried it myself, so you can only experiment like this when the RPi is already written off as a last resort.

**Reply by gghisellini** [reply]
I did everything I did with my Pi4 but with the Zero, and it's working!
I really don't know what was the issue with the Pi4.
I'll see to salvage it for another project.

Thank you so much for your help.

One last question: Is there a way of powering off the LEDs after there is no change in the image for X amount of time?
Or something like that?

**Reply by awawa-dev** [reply]
This is an automatic signal detection for the grabber, but it can only recognize one pre-recorded board: we usually record the grabber's board generated for the 'no HDMI signal' to turn off the LEDs (and optionally free up some grabber resources to save power). Point 4 in v17 https://www.hyperhdr.eu/2021/08/hyperhdr-v17beta-preview.html

**Reply by awawa-dev** [reply]
But it could be an interesting option for LED device drivers. If you want, you can create a new feature request here: https://github.com/awawa-dev/HyperHDR/issues

**Reply by gghisellini** [reply]
I created a new issue requesting the feature under https://github.com/awawa-dev/HyperHDR/issues/625
More than happy to know what you think about it!
